### PR TITLE
Split 835b488 part 2

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: include/easy_chat.h [hunk 1]
```diff
@@ -2,43 +2,48 @@
 #define GUARD_EASYCHAT_H
```
Comment (jiangzhengwenjz) [inline include/easy_chat.h]: GUARD_EASY_CHAT_H

File: src/berry.c [hunk 2]
```diff
@@ -3,9 +3,920 @@
...
+    {180, 500},
+    {200, 750},
+    {200, 750},
+    {150, 200}
+};
+
+// Leftover from R/S
+const struct BerryTree gBlankBerryTree = {};
+
+#define ENIGMA_BERRY_STRUCT ({const struct Berry2 *berries = (const struct Berry2 *)gBerries;berries[ITEM_ENIGMA_BERRY - FIRST_BERRY_INDEX];})
```
Comment (jiangzhengwenjz) [inline src/berry.c]: Inconsistent pointer space style with line 1077

File: src/mail.c [hunk 3]
```diff
@@ -70,8 +70,8 @@ struct MailViewResources {
     u8 monIconSpriteId;
     u8 unused;
     u8 mailArrangementType;
-    void (*copyEasyChatWord)(u8 *dest, u16 word);
-    void (*convertEasyChatWordsToString)(u8 *dest, const u16 *src, u16 length1, u16 length2);
+    u8 *(*copyEasyChatWord)(u8 *dest, u16 word);
+    u8 *(*convertEasyChatWordsToString)(u8 *dest, const u16 *src, u16 length1, u16 length2);
     const struct MailAttrStruct * messageLayout;
```
Comment (jiangzhengwenjz) [inline src/mail.c]: Inconsistent pointer space style with line 80/65

File: src/easy_chat.c [hunk 4]
```diff
@@ -0,0 +1,743 @@
...
+    for (i = 0; i < ARRAY_COUNT(sDeoxysValue); i++)
+    {
+        if (sDeoxysValue[i] == species)
+            return TRUE;
+    }
+
+    return FALSE;
+}
+
+static u8 IsWordUnlocked(u16 easyChatWord)
```
Comment (jiangzhengwenjz) [inline src/easy_chat.c]: bool8

File: src/easy_chat.c [hunk 5]
```diff
@@ -0,0 +1,743 @@
...
+        return TRUE;
+    case EC_GROUP_MOVE_1:
+    case EC_GROUP_MOVE_2:
+        return TRUE;
+    default:
+        return sEasyChatGroups[groupId].wordData.words[wordIndex].enabled;
+    }
+}
+
+static int EC_IsDeoxys(u16 species)
```
Comment (jiangzhengwenjz) [inline src/easy_chat.c]: can bool32 match the return type?

File: src/easy_chat.c [hunk 6]
```diff
@@ -0,0 +1,743 @@
...
+    s32 i;
+    u16 *ptr = sub_8143DA8();
+    for (i = 0; i < 4; i++)
+        ptr[i] = EC_WORD_UNDEFINED;
+}
+
+bool8 InitEasyChatSelection(void)
+{
+    sEasyChatSelectionData = Alloc(sizeof(*sEasyChatSelectionData));
+    if (!sEasyChatSelectionData)
```
Comment (jiangzhengwenjz) [inline src/easy_chat.c]: if (sEasyChatSelectionData == NULL)

File: src/easy_chat.c [hunk 7]
```diff
@@ -0,0 +1,743 @@
...
+static u8 IsWordUnlocked(u16 easyChatWord)
+{
+    u8 groupId = EC_GROUP(easyChatWord);
+    u32 index = EC_INDEX(easyChatWord);
+    if (!IsGroupSelectable(groupId))
+        return FALSE;
+    else
+        return UnlockedECMonOrMove(index, groupId);
+}
+
```
Comment (jiangzhengwenjz) [inline src/easy_chat.c]: redundant newline

File: src/pokemon_size_record.c [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+    memcpy(array, sGiftRibbonsMonDataIds, sizeof(sGiftRibbonsMonDataIds));
+
+    if (index < 11 && ribbonId < 65)
+    {
+        gSaveBlock1Ptr->giftRibbons[index] = ribbonId;
+        for (i = 0; i < PARTY_SIZE; i++)
+        {
+            struct Pokemon * mon = &gPlayerParty[i];
+
+            if (GetMonData(mon, MON_DATA_SPECIES) != 0 && GetMonData(mon, MON_DATA_SANITY_IS_EGG) == 0)
```
Comment (jiangzhengwenjz) [inline src/pokemon_size_record.c]: if (GetMonData(mon, MON_DATA_SPECIES) != SPECIES_NONE && !GetMonData(mon, MON_DATA_SANITY_IS_EGG))

File: data/bike.s [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+gUnknown_83E7D22::
+	.byte 4
+
+	.align 2
+gUnknown_83E7D24::
+	.4byte 1, 2, 15, 15, gUnknown_83E7D22, gUnknown_83E7D22, 1
+	.4byte 2, 2, 15, 15, gUnknown_83E7D22, gUnknown_83E7D22, 2
+	.4byte 3, 2, 15, 15, gUnknown_83E7D22, gUnknown_83E7D22, 3
+	.4byte 4, 2, 15, 15, gUnknown_83E7D22, gUnknown_83E7D22, 4
+
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply data/bike.s]: Not a big deal

File: data/naming_screen.s [hunk 10]
```diff
@@ -0,0 +1,159 @@
+#include "constants/maps.h"
+#include "constants/species.h"
+	.include "asm/macros.inc"
+	.include "constants/constants.inc"
+
+	.section .rodata
+	.align 2
+
+	.incbin "baserom.gba", 0x3E1800, 0xA80
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply data/naming_screen.s]: We'll cross that bridge when we get to it.

File: include/easy_chat.h (hunk 1 above)
Comment (PikalaxALT) [reply include/easy_chat.h]: No.

**Pull request merged**